src/components/Modal.vue

```diff
@@ -0,0 +1,274 @@
+		 */
+		title: {
+			type: String,
+			default: 'Título',
+		},
+		/**
+		 * Especifica o tamanho do modal. São 3 tamanhos implementados: 'sm', 'md', 'lg'.
+		 */
+		size: {
+			type: String,
+			default: 'md',
+		},
+		/**
+		 * Define o estado das ações do modal.
+		 */
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Controla a ação de fechar o modal ao clicar fora.
+		 */
+		noCloseOnBackdrop: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Controla a exibição do botão de fechar do modal.
+		 */
+		noCloseButton: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Controla a exibição do botão de cancelar do modal.
+		 */
+		noCancelButton: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Controla a exibição do rodapé (footer) do modal.
+		 */
+		noFooter: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Controla a exibição do cabeçalho (header) do modal.
+		 */
+		noHeader: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 *  Define texto do botão de ação do modal
+		 */
+		okButtonText: {
+			type: String,
+			default: 'Ok',
+		},
+		/**
+		 *  Define texto do botão de cancelar do modal
+		 */
+		cancelButtonText: {
+			type: String,
+			default: 'Cancelar',
+		},
+	},
+
+	mounted() {
+		this.internalShow = this.show;
+	},
+
+	watch: {
+		show(value) {
+			this.internalShow = value;
+		},
+	},
+
+	computed: {
+		maxBodyHeight() {
+			// 90% da largura subtraído o padding vertical (32 * 2) e subtraído o footer e o header
+			return `${ window.innerHeight * 0.9 - 32 * 2 - 110 }px`;
+		},
+	},
+
+	methods: {
+		close() {
+			/**
+			 * Evento que indica se o modal foi escondido.
+			* @event close
+			* @type {Event}
+			*/
+			this.$emit('close', true);
+			this.internalShow = !this.internalShow;
+		},
+		ok() {
+			/**
+			 * Evento que indica se o botão de ação do modal foi clicado.
+			* @event ok
+			* @type {Event}
+			*/
+			this.$emit('ok', true);
+			this.internalShow = !this.internalShow;
+		},
+	},
+};
+</script>
+<style lang="scss" scoped>
+@import '../assets/sass/app.scss';
+
+.cds-modal {
+	display: flex;
+	flex-direction: column;
+	max-height: 90%;
+	position: absolute;
+	right: calc(50% - 500px / 2);
+	width: 500px;
+	height: auto;
+	background-color: white;
+	padding: pYX(8, 7);
+	border-radius: $border-radius-medium;
+	box-shadow: 0px 0px 8px rgba(40, 90, 185, 0.2);
+	overflow-x: auto;
+	z-index: 999999999;
+
+	&__backdrop {
+		display: flex;
+		justify-content: center;
+		align-items: center;
+		position: fixed;
+		top: 0;
+		bottom: 0;
+		left: 0;
+		right: 0;
+		background-color: rgba(0, 0, 0, 0.45);
+		z-index: 99999999;
+	}
+
+	&__header {
+		display: flex;
+		justify-content: space-between;
+		padding: pb(7);
+
+		h3 {
+			@include subheading-1;
+			color: $n-900;
+		}
+	}
+
+	&__close-icon {
+		cursor: pointer;
+	}
+
+	&__footer {
+		display: flex;
+		justify-content: end;
+		margin-top: auto;
+		padding: pt(7);
+	}
+}
+
```

Comment:
Seria interessante colocar uma animação em `.cds-modal` para a abertura do modal, deixando-o mais suave. 
Algo como `animation: zoom .5s`
```suggestion
@keyframes zoom {
    0% {
        transform:scale(0);
    }
    100% {
        transform:scale(1);
    }
}
```